code.go

```diff
@@ -0,0 +1,260 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"sync"
+	"time"
+	"unsafe"
+)
+
+// 全局变量命名不规范
+var global_var int
+var GlobalVar2 string
+
+// 常量命名错误
+const max_SIZE = 100
+const minsize = 10
+
+// 1. 内存泄漏 - goroutine泄漏
+func memoryLeak() {
+	ch := make(chan int)
+	go func() {
+		for {
+			select {
+			case <-ch:
+				return
+			default:
+				// 这个goroutine永远不会退出
+				time.Sleep(time.Second)
+			}
+		}
+	}()
+	// 忘记关闭channel
+}
+
+// 2. 竞态条件
+var counter int
+
+func raceCondition() {
+	for i := 0; i < 1000; i++ {
+		go func() {
+			counter++ // 没有锁保护
+		}()
+	}
+}
+
+// 3. 错误的错误处理
+func badErrorHandling() {
+	file, err := os.Open("nonexistent.txt")
+	if err != nil {
+		// 错误：忽略错误
+		return
+	}
+	defer file.Close()
+
+	data := make([]byte, 100)
+	n, err := file.Read(data)
+	// 错误：没有检查错误
+	fmt.Println("Read", n, "bytes")
+}
+
+// 4. 资源泄漏
+func resourceLeak() {
+	file, _ := os.Open("test.txt") // 忽略错误
+	// 错误：没有defer close
+	data := make([]byte, 100)
+	file.Read(data)
+	// file永远不会被关闭
+}
+
+// 5. 切片越界和nil指针
+func sliceAndNilErrors() {
+	var slice []int
+	fmt.Println(slice[0]) // panic: 空切片访问
+
+	var ptr *int
+	fmt.Println(*ptr) // panic: nil指针解引用
+
+	arr := []int{1, 2, 3}
+	fmt.Println(arr[10]) // panic: 索引越界
```

Comment:
Check array bounds before accessing to prevent panic. Index out of bounds causes runtime panic.

```suggestion
	if len(arr) > 10 {
		fmt.Println(arr[10])
	}
```